src/routes/metadata.ts

```diff
@@ -215,46 +272,52 @@ router.post(
   }
 );
 
-// data must be array of arrays with each array lesser than MAX_BATCH_SIZE
-async function insertDataInBatchForTable(tableName, data) {
-  return knexWrite.transaction(async (trx) => {
-    for (const batch of data) {
-      // eslint-disable-next-line no-await-in-loop
-      await knexWrite(tableName).insert(batch).transacting(trx);
-    }
-  });
-}
-
 if (process.env.NODE_ENV === "development") {
   // API for dev env only to test if v1 continue to work after deploying v2
-  router.post("/set_nonce", validationMiddleware(["pub_key_X", "pub_key_Y"]), validateNamespace, async (req, res) => {
-    try {
-      const { pub_key_X: pubKeyX, pub_key_Y: pubKeyY, tableName } = req.body;
+  router.post(
+    "/set_nonce",
+    celebrate({
+      [Segments.BODY]: Joi.object({
+        namespace: Joi.string().max(128),
+        pub_key_X: Joi.string().length(64).required(),
+        pub_key_Y: Joi.string().length(64).required(),
+      }),
+    }),
+    validateNamespace,
+    async (req, res) => {
+      try {
+        const { pub_key_X: pubKeyX, pub_key_Y: pubKeyY, tableName }: { pub_key_X: string; pub_key_Y: string; tableName: string } = req.body;
 
-      const key = constructKey(pubKeyX, pubKeyY, NAMESPACES.nonceV2);
+        const key = constructKey(pubKeyX, pubKeyY, NAMESPACES.nonceV2);
 
-      await knexWrite(tableName).insert({
-        key,
-        value: "<v1>",
-      });
+        await knexWrite(tableName).insert({
+          key,
+          value: "<v1>",
+        });
 
-      try {
-        await redis.setEx(key, REDIS_TIMEOUT, "<v1>");
+        try {
+          await redis.setEx(key, REDIS_TIMEOUT, "<v1>");
+        } catch (error) {
+          log.warn("redis set failed", error);
+        }
+
+        return res.json({});
       } catch (error) {
-        log.warn("redis set failed", error);
+        log.error("set_nonce failed", error);
+        return res.status(500).json({ error: getError(error), success: false });
       }
-
-      return res.json({});
-    } catch (error) {
-      log.error("set_nonce failed", error);
-      return res.status(500).json({ error: getError(error), success: false });
     }
-  });
+  );
 }
 
 router.post(
   "/get_or_set_nonce",
-  validationMiddleware(["pub_key_X", "pub_key_Y"]),
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      pub_key_X: Joi.string().length(64).required(),
```

Comment:
set_data and namespace are missing in validations.

Reply:
fixed